ql/src/java/org/apache/hadoop/hive/ql/ddl/view/create/AbstractCreateViewAnalyzer.java

```diff
@@ -0,0 +1,120 @@
+    return schema;
+  }
+
+  protected SemanticAnalyzer analyzeQuery(ASTNode select, String fqViewName) throws SemanticException {
+    QueryState innerQueryState = new QueryState.Builder().withHiveConf(conf).build();
+    innerQueryState.getConf().setBoolVar(HiveConf.ConfVars.HIVE_RESULTSET_USE_UNIQUE_COLUMN_NAMES, false);
+
+    SemanticAnalyzer analyzer = (SemanticAnalyzer) SemanticAnalyzerFactory.get(innerQueryState, select);
+    ctx.setEnableUnparse(true);
+    analyzer.forViewCreation(fqViewName);
+    analyzer.analyze(select, ctx);
+    analyzer.executeUnparseTranlations();
+
+    queryState.setLineageState(innerQueryState.getLineageState());
+    queryState.getLineageState().mapDirToOp(new Path(fqViewName), analyzer.getSinkOp());
+
+    addInputs(analyzer);
+
+    return analyzer;
+  }
+
+  private void addInputs(SemanticAnalyzer analyzer) {
+    inputs.addAll(analyzer.getInputs());
+    for (Map.Entry<String, TableScanOperator> entry : analyzer.getTopOps().entrySet()) {
+      String alias = entry.getKey();
+      TableScanOperator topOp = entry.getValue();
+      ReadEntity parentViewInfo = PlanUtils.getParentViewInfo(alias, analyzer.getViewAliasToInput());
+
+      // Adds tables only for create view (PPD filter can be appended by outer query)
+      Table table = topOp.getConf().getTableMetadata();
+      PlanUtils.addInput(inputs, new ReadEntity(table, parentViewInfo));
+    }
+  }
+
+  protected void validateTablesUsed(SemanticAnalyzer analyzer) throws SemanticException {
+    // Do not allow view to be defined on temp table or other materialized view
+    Set<String> tableAliases = analyzer.getQB().getTabAliases();
+    for (String alias : tableAliases) {
+      try {
+        if (SemanticAnalyzer.DUMMY_TABLE.equals(alias)) {
+          continue;
+        }
+        Table table = analyzer.getTableObjectByName(analyzer.getQB().getTabNameForAlias(alias));
+        if (table.isTemporary()) {
+          throw new SemanticException("View definition references temporary table " + alias);
+        }
+        if (table.isMaterializedView()) {
+          throw new SemanticException("View definition references materialized view " + alias);
+        }
+      } catch (HiveException ex) {
+        throw new SemanticException(ex);
+      }
+    }
+  }
+
+  protected void validateReplaceWithPartitions(String viewName, Table oldView, List<FieldSchema> partitionColumns)
+      throws SemanticException {
+    String partitionViewErrorMsg = "The following view has partition, it could not be replaced: " + viewName;
+    try {
+      if (!oldView.getPartCols().isEmpty() && !db.getPartitions(oldView).isEmpty() &&
+          !oldView.getPartCols().equals(partitionColumns)) {
+        throw new SemanticException(ErrorMsg.REPLACE_VIEW_WITH_PARTITION.getMsg(partitionViewErrorMsg));
+      }
+    } catch (HiveException e) {
+      throw new SemanticException(ErrorMsg.REPLACE_VIEW_WITH_PARTITION.getMsg(partitionViewErrorMsg));
```

Comment:
Again. this `SemanticException` will be double-wrapped

Reply:
I agree, fixed it.